experiment/config-forker/main.go

```diff
@@ -0,0 +1,317 @@
+	if args == nil {
+		return nil, nil
+	}
+	if replacements == "" {
+		return args, nil
+	}
+
+	var rs []string
+	as := strings.Split(replacements, ", ")
+	for _, r := range as {
+		s := strings.Split(r, " -> ")
+		if len(s) != 2 {
+			return nil, fmt.Errorf("failed to parse replacement %q", r)
+		}
+		v, err := evaluateTemplate(s[1], struct{ Version string }{version})
+		if err != nil {
+			return nil, err
+		}
+		rs = append(rs, s[0], v)
+	}
+	replacer := strings.NewReplacer(rs...)
+
+	newArgs := make([]string, 0, len(args))
+	for _, a := range args {
+		newArgs = append(newArgs, replacer.Replace(a))
+	}
+
+	return newArgs, nil
+}
+
+func fixImage(image, version string) string {
+	return strings.ReplaceAll(image, "-master", "-"+version)
+}
+
+func fixBootstrapArgs(args []string, version string) []string {
+	if args == nil {
+		return nil
+	}
+	replacer := strings.NewReplacer(
+		"--repo=k8s.io/kubernetes=master", "--repo=k8s.io/kubernetes=release-"+version,
+		"--repo=k8s.io/kubernetes", "--repo=k8s.io/kubernetes=release-"+version,
+		"--branch=master", "--branch=release-"+version,
+	)
+	newArgs := make([]string, 0, len(args))
+	for _, arg := range args {
+		newArgs = append(newArgs, replacer.Replace(arg))
+	}
+	return newArgs
+}
+
+func fixExtraRefs(refs []prowapi.Refs, version string) []prowapi.Refs {
+	if refs == nil {
+		return nil
+	}
+	newRefs := make([]prowapi.Refs, 0, len(refs))
+	for _, r := range refs {
+		if r.Org == "kubernetes" && r.Repo == "kubernetes" && r.BaseRef == "master" {
+			r.BaseRef = "release-" + version
+		}
+		newRefs = append(newRefs, r)
+	}
+	return newRefs
+}
+
+func fixEnvVars(vars []v1.EnvVar, version string) []v1.EnvVar {
+	if vars == nil {
+		return nil
+	}
+	newVars := make([]v1.EnvVar, 0, len(vars))
+	for _, v := range vars {
+		if strings.Contains(strings.ToUpper(v.Name), "BRANCH") && v.Value == "master" {
+			v.Value = "release-" + version
+		}
+		newVars = append(newVars, v)
+	}
+	return newVars
+}
+
+func generateNameVariant(name, version string) string {
+	nameVersion := strings.ReplaceAll(version, ".", "-")
+	if !strings.HasSuffix(name, "-master") {
+		return name + "-" + nameVersion
+	}
+	return strings.ReplaceAll(name, "-master", "-"+nameVersion)
+}
+
+type options struct {
+	jobConfig  string
+	outputPath string
+	newVersion string
+}
+
+func parseFlags() options {
+	o := options{}
+	flag.StringVar(&o.jobConfig, "job-config", "", "Path to the job config")
+	flag.StringVar(&o.outputPath, "output", "", "Path to the output yaml. if not specified, just validate.")
+	flag.StringVar(&o.newVersion, "version", "", "Version number to generate jobs for")
+	flag.Parse()
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.jobConfig == "" {
+		return errors.New("--job-config must be specified")
+	}
+	if o.newVersion == "" {
+		return errors.New("--version must be specified")
+	}
+	if match, err := regexp.MatchString(`^\d+.\d+$`, o.newVersion); err != nil || !match {
```

Comment:
I think you meant `\.` and not `.` (wildcard)

Reply:
Fixed.